### PR TITLE
[.NET 8 in-proc] Multi targeting .NET 6 and .NET 8

### LINKED ISSUE #10044: Multi-targeting host build to create .NET 8 artifacts
Update the in-proc host to add a .NET 8 target.

Feature branch: https://github.com/Azure/azure-functions-host/tree/feature/in-proc8

```[tasklist]
### Tasks
- [x] Review and merge PR #9858
- [x] Create feature branch (from in-proc) for new .NET 8 changes
- [x] Backport 9858 to in-proc (PR #10079)
- [x] Update runtime assembly files in dev and in-proc branch
- [x] Address any issues and failures in .NET 8 in-proc branch
```

```[tasklist]
### Test Failures
- [x] DependencyTests - Need to update file path (different test for net6.0 vs net8.0)
- [x] RpcWorkerConfigFactoryTests - Unexpected process count (my machines core count is 16 which is impacting this test, might be fine in the CI pipeline)
- [x] ProxyFunctionDescriptorProviderTests - Call is ambiguous between the following methods or properties
- [x] ProxyFunctionDescriptorTests - Call is ambiguous between the following methods or properties
- [x] WebFunctionsManagerTests - MoqException: IConfiguration
- [x] DefaultDependencyValidatorTests - DryIoc.ContainerException
- [x] LoggingConfigurationTests - DryIoc.ContainerException
- [x] NullReference and ArgumentNull exceptions
- [x] TFM error
- [x] FunctionAssemblyLoadContextEndToEndTests failures
- [x] Missing debug_net6.0 dir in pipeline
```

### PR DESCRIPTION
**Pull request by fabiocav**
### Issue describing the changes in this PR

resolves #10044 

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [ ] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [ ] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "6.0.419",
+    "version": "8.0.204",
```
Comment (jviau) [inline global.json]: Dev uses `8.0.101`. 
```suggestion
    "version": "8.0.101",
```

File: src/WebJobs.Script.Abstractions/WebJobs.Script.Abstractions.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <Import Project="..\..\build\common.props" />
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
```
Comment (jviau) [inline src/WebJobs.Script.Abstractions/WebJobs.Script.Abstractions.csproj]: We shouldn't need multi-target semantics in a single TFM project
```suggestion
    <TargetFramework>netstandard2.0</TargetFramework>
```

File: tools/ExtensionsMetadataGenerator/test/ExtensionsMetadataGeneratorTests/ExtensionsMetadataGeneratorTests.csproj [hunk 3]
```diff
@@ -7,7 +7,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <None Remove="ExistingRuntimeAssemblies.txt" />
+    <None Remove="ExistingRuntimeAssemblies-net6.txt" />
```
Comment (jviau) [inline tools/ExtensionsMetadataGenerator/test/ExtensionsMetadataGeneratorTests/ExtensionsMetadataGeneratorTests.csproj]:
```suggestion
    <None Remove="ExistingRuntimeAssemblies-*.txt" />
```

Can remove the 2nd line then

File: tools/ExtensionsMetadataGenerator/test/ExtensionsMetadataGeneratorTests/ExtensionsMetadataGeneratorTests.csproj [hunk 4]
```diff
@@ -37,7 +38,10 @@
   </ItemGroup>
 
   <ItemGroup>
-    <EmbeddedResource Include="ExistingRuntimeAssemblies.txt">
+    <EmbeddedResource Include="ExistingRuntimeAssemblies-net8.txt">
```
Comment (jviau) [inline tools/ExtensionsMetadataGenerator/test/ExtensionsMetadataGeneratorTests/ExtensionsMetadataGeneratorTests.csproj]: Can use globbing here

```suggestion
    <EmbeddedResource Include="ExistingRuntimeAssemblies-*.txt">
```

File: src/WebJobs.Script.Abstractions/WebJobs.Script.Abstractions.csproj (hunk 2 above)
Comment (fabiocav) [reply src/WebJobs.Script.Abstractions/WebJobs.Script.Abstractions.csproj]: I believe this was in place to override what was defined in `common.props`, but common has been changed and no longer defines a target, so we should be OK to change (I think @liliankasem went through some of the projects to fix this up, but this was likely missed)

**Review by liliankasem — APPROVED** [review]

**Review by brettsam — APPROVED** [review]

**Pull request merged**